[2025-08-09T17:11:51.179+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: ml_pipeline.model_training manual__2025-08-09T17:06:00.445445+00:00 [queued]>
[2025-08-09T17:11:51.200+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: ml_pipeline.model_training manual__2025-08-09T17:06:00.445445+00:00 [queued]>
[2025-08-09T17:11:51.202+0000] {taskinstance.py:2170} INFO - Starting attempt 2 of 2
[2025-08-09T17:11:51.232+0000] {taskinstance.py:2191} INFO - Executing <Task(PythonOperator): model_training> on 2025-08-09 17:06:00.445445+00:00
[2025-08-09T17:11:51.243+0000] {standard_task_runner.py:60} INFO - Started process 1806 to run task
[2025-08-09T17:11:51.249+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'ml_pipeline', 'model_training', 'manual__2025-08-09T17:06:00.445445+00:00', '--job-id', '64', '--raw', '--subdir', 'DAGS_FOLDER/ml_pipeline.py', '--cfg-path', '/tmp/tmpo0sxd8hx']
[2025-08-09T17:11:51.253+0000] {standard_task_runner.py:88} INFO - Job 64: Subtask model_training
[2025-08-09T17:11:51.340+0000] {task_command.py:423} INFO - Running <TaskInstance: ml_pipeline.model_training manual__2025-08-09T17:06:00.445445+00:00 [running]> on host 1a0a448a8cb5
[2025-08-09T17:11:51.441+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='sarah' AIRFLOW_CTX_DAG_ID='ml_pipeline' AIRFLOW_CTX_TASK_ID='model_training' AIRFLOW_CTX_EXECUTION_DATE='2025-08-09T17:06:00.445445+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-08-09T17:06:00.445445+00:00'
[2025-08-09T17:12:30.495+0000] {logging_mixin.py:188} INFO - STDOUT:
[2025-08-09T17:12:30.498+0000] {logging_mixin.py:188} INFO -  Fitting 3 folds for each of 10 candidates, totalling 30 fits
üèÉ View run XGBoost_RandSearch at: https://dagshub.com/SamMebarek/mlops-som.mlflow/#/experiments/0/runs/52777eb275014a4f9f214539c7d064a6
üß™ View experiment at: https://dagshub.com/SamMebarek/mlops-som.mlflow/#/experiments/0
[2025-08-09T17:12:30.499+0000] {logging_mixin.py:188} INFO - STDERR:
[2025-08-09T17:12:30.500+0000] {logging_mixin.py:188} INFO -  2025-08-09 17:11:54,825 - training.utils.common - INFO - YAML charg√© : /opt/***/project/config/config.yaml
2025-08-09 17:11:54,832 - training.utils.common - INFO - YAML charg√© : /opt/***/project/config/params.yaml
2025-08-09 17:11:54,834 - training.utils.common - INFO - R√©pertoire cr√©√© ou existant : /opt/***/project/models
2025-08-09 17:11:54,834 - training.utils.common - INFO - R√©pertoire cr√©√© ou existant : /opt/***/project/models
2025-08-09 17:11:55,859 - __main__ - INFO - Loaded processed data: /opt/***/project/data/processed/clean_data.csv (shape=(108060, 14))
2025-08-09 17:11:55,884 - __main__ - INFO - Split data: train_shape=(86448, 11), test_shape=(21612, 11)
2025/08/09 17:11:55 WARNING mlflow.utils.git_utils: Failed to import Git (the Git executable is probably not on your PATH), so Git SHA is not available. Error: Failed to initialize: Bad git executable.
The git executable must be specified in one of the following ways:
    - be included in your $PATH
    - be set via $GIT_PYTHON_GIT_EXECUTABLE
    - explicitly set via git.refresh(<full-path-to-git-executable>)

All git commands will error until this is rectified.

This initial message can be silenced or aggravated in the future by setting the
$GIT_PYTHON_REFRESH environment variable. Use one of the following values:
    - quiet|q|silence|s|silent|none|n|0: for no message or exception
    - warn|w|warning|log|l|1: for a warning message (logging level CRITICAL, displayed by default)
    - error|e|exception|raise|r|2: for a raised exception

Example:
    export GIT_PYTHON_REFRESH=quiet

2025-08-09 17:11:56,232 - __main__ - INFO - Starting hyperparameter search and training.
2025-08-09 17:12:14,698 - __main__ - INFO - Best params: {'colsample_bytree': 0.9219644132807024, 'gamma': 0.06692515264078093, 'learning_rate': 0.19307895078298942, 'max_depth': 5, 'n_estimators': 86, 'subsample': 0.6271914913054505}
2025-08-09 17:12:14,698 - __main__ - INFO - R^2 score: 0.9998
2025/08/09 17:12:28 INFO mlflow.store.model_registry.abstract_store: Waiting up to 300 seconds for model version to finish creation. Model name: PriceFixerModel, version 45
2025-08-09 17:12:28,144 - __main__ - INFO - Registered model 'PriceFixerModel' from source=mlflow-artifacts:/4829b4e5ce254fea8f678e40a4133671/52777eb275014a4f9f214539c7d064a6/artifacts/xgb_model, run_id=52777eb275014a4f9f214539c7d064a6
2025-08-09 17:12:28,661 - training.repository.repository - INFO - Mod√®le sauvegard√© ‚Üí /opt/***/project/models/xgb_model.pkl
2025-08-09 17:12:28,699 - __main__ - INFO - Local MLflow dir: models/xgb_model_mlflow_manual__2025-08-09T17:06:00.445445+00:00 (size=244004 bytes)
2025-08-09 17:12:28,699 - __main__ - INFO - Model saved locally at: /opt/***/project/models/xgb_model.pkl
2025-08-09 17:12:28,702 - __main__ - INFO - Pushed metrics to Pushgateway (http://pushgateway:9091) status=1 duration_s=33.892 rows=108060 r2=0.9997696380635521 artifact_bytes=484994
[2025-08-09T17:12:30.501+0000] {python.py:201} INFO - Done. Returned value was: None
[2025-08-09T17:12:30.518+0000] {taskinstance.py:1138} INFO - Marking task as SUCCESS. dag_id=ml_pipeline, task_id=model_training, execution_date=20250809T170600, start_date=20250809T171151, end_date=20250809T171230
[2025-08-09T17:12:30.568+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 0
[2025-08-09T17:12:30.608+0000] {taskinstance.py:3280} INFO - 1 downstream tasks scheduled from follow-on schedule check
